src/pkg/services/m365/api/channels_pager.go

```diff
@@ -142,15 +151,15 @@ func (c Channels) GetChannelMessageIDs(
 	teamID, channelID, prevDeltaLink string,
 	canMakeDeltaQueries bool,
 ) ([]string, []string, DeltaUpdate, error) {
-	added, removed, du, err := getAddedAndRemovedItemIDs(
+	added, _, removed, du, err := getAddedAndRemovedItemIDs[models.ChatMessageable](
 		ctx,
 		c.NewChannelMessagePager(teamID, channelID),
 		c.NewChannelMessageDeltaPager(teamID, channelID, prevDeltaLink),
 		prevDeltaLink,
 		canMakeDeltaQueries,
-		addedAndRemovedByDeletedDateTime)
+		addedAndRemovedByDeletedDateTime[models.ChatMessageable])
 
-	return added, removed, du, clues.Stack(err).OrNil()
+	return maps.Keys(added), removed, du, clues.Stack(err).OrNil()
```

Comment:
To maintain consistency across implementations, can this please switch to a `map[string]time.Time` return value?